### DISCUSSION TITLE
Suggested branches/revisions (for branch-reset target, in particular)

### DISCUSSION THREAD

**Discussion by ingolohmar** [magit/magit#5042] [Ideas]
I often find myself resetting non-current ("other") branches to a particular revision.  In a log display, I do "b x", choose the other branch, and then I would like to choose a raw commit the branch should point to (ie, *not* another branch or  tag).

I would love to have the minibuffer defaults include the commit on which point is, but could not find a way to do that (so I do a `C-w` first, and then paste into the second prompt).  Is there such a way, or is this more generally interesting (offering commit on point as one of the history defaults)?  I have seen one related discussion for defaulting a diff command to a selected range.

Otherwies I would have to hack magit's internals (would need to have a closer look) to achieve the behavior I want.

**Comment by tarsius** [comment]
I've added it as the first "previous history element", reachable with `M-p`.

**Reply by tarsius** [reply]
By the way, thanks for reminding me of that command. I should use it more, and once I done so for a while, I might end up tweaking it a bit.

**Reply by tarsius** [reply]
... and it has begun: 2b02bfb0495bf738e006d82f3a233e8f511b90b8. :grinning:

**Reply by ingolohmar** [reply]
Wow, that's awesome!  Will soon try it out, thanks.
